src/chain_sync/sync_status.rs

```diff
@@ -0,0 +1,246 @@
+    /// Can be negative if the node is slightly ahead, due to estimation variance.
+    epochs_behind: i64,
+    /// List of active fork synchronization tasks the node is currently handling.
+    active_forks: Vec<ForkSyncInfo>,
+    /// When the node process started.
+    node_start_time: DateTime<Utc>,
+    /// Last time this status report was generated.
+    last_updated: DateTime<Utc>,
+}
+
+lotus_json_with_self!(SyncStatusReport);
+
+impl SyncStatusReport {
+    pub(crate) fn init() -> Self {
+        Self {
+            node_start_time: Utc::now(),
+            ..Default::default()
+        }
+    }
+
+    pub(crate) fn set_current_chain_head_key(&mut self, tipset_key: TipsetKey) {
+        self.current_head_key = Some(tipset_key);
+    }
+
+    pub(crate) fn set_current_chain_head_epoch(&mut self, epoch: ChainEpoch) {
+        self.current_head_epoch = epoch;
+    }
+
+    pub(crate) fn get_current_chain_head_epoch(&self) -> ChainEpoch {
+        self.current_head_epoch
+    }
+
+    pub(crate) fn set_network_head(&mut self, epoch: ChainEpoch) {
+        self.network_head_epoch = epoch;
+    }
+
+    pub(crate) fn get_network_head_epoch(&self) -> ChainEpoch {
+        self.network_head_epoch
+    }
+
+    pub(crate) fn get_current_chain_head_key(&self) -> Option<&TipsetKey> {
+        self.current_head_key.as_ref()
+    }
+
+    pub(crate) fn set_epochs_behind(&mut self, epochs_behind: i64) {
+        self.epochs_behind = epochs_behind;
+    }
+
+    pub(crate) fn get_epochs_behind(&self) -> i64 {
+        self.epochs_behind
+    }
+
+    pub(crate) fn set_status(&mut self, status: NodeSyncStatus) {
+        self.status = status;
+    }
+
+    pub(crate) fn get_status(&self) -> NodeSyncStatus {
+        self.status
+    }
+
+    pub(crate) fn update_active_forks(&mut self, active_forks: Vec<ForkSyncInfo>) {
+        self.active_forks = active_forks;
+    }
+
+    pub(crate) fn get_active_forks(&self) -> &Vec<ForkSyncInfo> {
+        &self.active_forks
+    }
+
+    pub fn get_last_updated(&self) -> &DateTime<Utc> {
+        &self.last_updated
+    }
+
+    pub(crate) fn update<DB: Blockstore + Sync + Send + 'static>(
+        &mut self,
+        state_manager: &Arc<StateManager<DB>>,
+        current_active_forks: Vec<ForkSyncInfo>,
+        stateless_mode: bool,
+    ) {
+        let heaviest = state_manager.chain_store().heaviest_tipset();
+        let current_chain_head_epoch = heaviest.epoch();
+        self.set_current_chain_head_key(heaviest.key().clone());
+        self.set_current_chain_head_epoch(current_chain_head_epoch);
+
+        let now = Utc::now();
+        let now_ts = now.timestamp() as u64;
+        let seconds_per_epoch = state_manager.chain_config().block_delay_secs;
+        let network_head_epoch = calculate_expected_epoch(
+            now_ts,
+            state_manager.chain_store().genesis_block_header().timestamp,
+            seconds_per_epoch,
+        );
```

Comment:
I think this could be a separate method somewhere.

Reply:
Everywhere else we are directly using the `calculate_expected_epoch` and passing the fields, but here we have introduced the variables just so we can reuse them. I don't think we need to introduce a separate method just for this specific use case.